Documentation/Decode-Enums.md

```diff
@@ -0,0 +1,69 @@
+# Decoding Enums
+
+Decoding Structs and Classes is a similar process, but Enums can mix things up.
+If your Enum inherits from a Raw Type `String` or `Int`, then making it conform
+to `Decodable` is as simple as pie. Let's look at a `Role` type enum:
+
+```swift
+enum Role: String {
+  case User 
+  case Admin
+  case SuperAdmin
+}
+```
+
+To make `Role` conform to `Decodable`, use this one line:
+
+```swift
+extension Role: Decodable { }
+```
+
+"THAT'S IT?! How?", you ask. Enums with a raw type like `String` or `Int`
+conform to `RawRepresentable` and we added a default [implementation] for these
+cases. These types of enums can be assigned the raw value that represents each
+case or Swift will automatically assign a raw value for you. In this instance,
+each case has a raw value of a `String` that is that case name, like `case
+User = "User"`. With an `Int`, Swift would automatically assign the top case to
+`0` and increment by one for each case afterward.
+
+[implementation]: ../Argo/Extensions/RawRepresentable.swift
+
+So, what if the enum doesn't have a raw type? Conforming to `Decodable` is a
+bit more involved but not too hard. Let's look at an example:
+
+```swift
+enum FootRace {
+  case HalfMarathon
+  case Marathon
+  case UltraMarathon
+}
+```
+
+We can write `Decodable` for `FootRace` like so:
+
+```swift
+extension FootRace: Decodable {
+  static func decode(j: JSON) -> Decoded<FootRace> {
+    switch j {
+
+    // First, make sure JSON is a number.
+    case let .Number(n):
```

Comment:
how about naming this `distance` instead of `n`. I think that will make things clearer below without the need for comments.